### PR TITLE
improvements sticky_labor 

### PR DESCRIPTION
**Pull request by deleip**
## :bird: Description of this PR :bird:

This PR changes the labor share target in `sticky_labor` from a regional constraint  to a cellular constraint, and improves the start values of `v38_capital_need` and  `v38_laborhours_need`. This speeds up scenarios with labor share target and allows the high resolution output script to run through.
In addition `nl_fix`, `nl_relax`, and `nl_release` were added, such that the lp_nlp optimization can be used also for the sticky_labor realization.

Unfortunately the high res output script still doesn't find a feasible solution for the full FSDP.

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - FSEC CapitalSubst develop branch: 2h 54mins
  - FSEC CapitalSubst this PR: 1h 57mins
  - FSEC CapitalSubst this PR with lp_nlp: 1h 34mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [x] content review done (at least 1)
- [x] RSE review done (at least 1)

### CODE REVIEW

File: modules/38_factor_costs/sticky_labor/nl_fix.gms [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+* ### nl_fix ###
+
+v38_laborhours_need.fx(j,kcr) =  v38_laborhours_need.l(j,kcr);
+v38_capital_need.fx(j,kcr,mobil38) =  v38_capital_need.l(j,kcr,mobil38);
+
+* release v38_relax_CES_lp for linear version of MAgPIE
+v38_relax_CES_lp.l(j,kcr) = 0;
+v38_relax_CES_lp.lo(j,kcr) = -Inf;
+v38_relax_CES_lp.up(j,kcr) = Inf;
```
Comment (flohump) [inline modules/38_factor_costs/sticky_labor/nl_fix.gms:16]: Add new line at end of file?

File: modules/38_factor_costs/sticky_labor/nl_relax.gms [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+*** |  of MAgPIE and licensed under AGPL-3.0-or-later. Under Section 7 of
+*** |  AGPL-3.0, you are granted additional permissions described in the
+*** |  MAgPIE License Exception, version 1.0 (see LICENSE file).
+*** |  Contact: magpie@pik-potsdam.de
+
+* ### nl_relax ###
+
+v38_laborhours_need.l(j,kcr) = v38_laborhours_need.l(j,kcr) * 1.1;
+v38_capital_need.l(j,kcr,"immobile") = sum(cell(i,j), (i38_ces_shr(j,kcr) / (i38_ces_scale(j,kcr)**s38_ces_elast_par - (1-i38_ces_shr(j,kcr))*(pm_labor_prod(t,j) * pm_productivity_gain_from_wages(t,i)*v38_laborhours_need.l(j,kcr)**(-s38_ces_elast_par))))**(1/s38_ces_elast_par)) * s38_immobile;
+v38_capital_need.l(j,kcr,"mobile") = v38_capital_need.l(j,kcr,"immobile") * (1-s38_immobile) / s38_immobile;
```
Comment (flohump) [inline modules/38_factor_costs/sticky_labor/nl_relax.gms:12]: new line?

File: modules/38_factor_costs/sticky_labor/nl_release.gms [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+* ### nl_release ###
+
+v38_laborhours_need.lo(j,kcr) = 0.1 * v38_laborhours_need.l(j,kcr);
+v38_laborhours_need.up(j,kcr) = 10 * v38_laborhours_need.l(j,kcr);
+v38_capital_need.lo(j,kcr,mobil38) = 0.1 * v38_capital_need.l(j,kcr,mobil38);
+v38_capital_need.up(j,kcr,mobil38) = 10 * v38_capital_need.l(j,kcr,mobil38);
+
+* fix v38_relax_CES_lp to zero for non-linear version of MAgPIE
+v38_relax_CES_lp.fx(j,kcr) = 0;
```
Comment (flohump) [inline modules/38_factor_costs/sticky_labor/nl_release.gms:16]: new line?

File: modules/38_factor_costs/sticky_labor/presolve.gms [hunk 4]
```diff
@@ -21,43 +21,60 @@ p38_labor_need(t,i,kcr) = i38_fac_req(t,i,kcr)  * pm_cost_share_crops(t,i,"labor
...
+v38_laborhours_need.lo(j,kcr) = 0.1 * v38_laborhours_need.l(j,kcr);
+v38_laborhours_need.up(j,kcr) = 10 * v38_laborhours_need.l(j,kcr);
+
+* fix capital need until startyear of labor substitution
+if (m_year(t) <= s38_startyear_labor_substitution,
+  v38_capital_need.fx(j,kcr,mobil38) = sum(cell(i,j), p38_capital_need(t,i,kcr,mobil38));
+else 
+* in case of labor share target: initialize level by assuming that the regional target shares would be enforced in each cell
+  if (s38_target_labor_share <> 0,
+*    v38_capital_need.l(j,kcr,"immobile") = sum(cell(i,j), (i38_ces_shr(j,kcr) / (i38_ces_scale(j,kcr)**s38_ces_elast_par - (1-i38_ces_shr(j,kcr))*(pm_labor_prod(t,j) * pm_productivity_gain_from_wages(t,i)*v38_laborhours_need.l(j,kcr)**(-s38_ces_elast_par))))**(1/s38_ces_elast_par)) * s38_immobile;
```
Comment (flohump) [inline modules/38_factor_costs/sticky_labor/presolve.gms:72]: Delete this line if commented out?

File: modules/38_factor_costs/sticky_labor/presolve.gms [hunk 5]
```diff
@@ -76,3 +93,6 @@ else
     p38_capital_mobile(t,j)=p38_capital_mobile(t,j)*(1-s38_depreciation_rate)**(m_timestep_length);
 
     );
+
+* fix v38_relax_CES_lp to zero for non-linear version of MAgPIE
+v38_relax_CES_lp.fx(j,kcr) = 0;
```
Comment (flohump) [inline modules/38_factor_costs/sticky_labor/presolve.gms:98]: new line?

**Review by flohump — APPROVED** [review]
Thanks for improving this realization and also for adding the LP-NLP files.
The changes looks sound to me.

**Review by tscheypidi — APPROVED** [review]
looks good and impressive performance improvements

**Pull request merged**